app/assets/javascripts/components/story/CollapsedStory/index.js

```diff
@@ -23,39 +27,177 @@ const storyClassName = (story, additionalClassname = '') => {
       'Story--estimated': !isStoryNotEstimated
     },
     additionalClassname
-  );
-};
-
-export const CollapsedStory = ({ onToggle, story, updateCollapsedStory, project, className, title }) =>
-  <div
-    className={storyClassName(story, className)}
-    onClick={onToggle}
-    title={title}
-  >
-    <StoryPopover story={story}>
-      <div className='Story__icons-block'>
-        <StoryIcon storyType={story.storyType} />
-        <CollapsedStoryEstimate estimate={story.estimate} />
-        <StoryDescriptionIcon description={story.description} />
-      </div>
-    </StoryPopover>
-
-    <CollapsedStoryInfo story={story} />
-
-    <CollapsedStoryStateActions story={story}
-      onUpdate={(newAttributes) =>
-        updateCollapsedStory(story.id, project.id, newAttributes)}
-    />
-  </div>
+  )
+}
+
+const type = {
+  story: 'STORY'
+}
+
+const calculatePosition = (aboveStory, bellowStory, storiesArray, index) => {
+  if (bellowStory === undefined) {
+    return (Number(storiesArray[index].position) + 1)
+  }else if (aboveStory !== undefined && bellowStory !== undefined) {
+    return (Number(bellowStory.position) + Number(aboveStory.position)) / 2;
+  }else if (aboveStory === undefined) {
+    return (Number(storiesArray[index].position) - 1);
+  }
+}
```

Comment:
I believe you could improve these `if`s. For example, why in the second `if` you are checking if `aboveStory` is not `undefined`? You already checked that is the first `if` ;)

```js
const calculatePosition = (aboveStory, bellowStory, storiesArray, index) => {
  if (bellowStory === undefined) {
    return Number(storiesArray[index].position) + 1;
  } else if (aboveStory === undefined) {
    return Number(storiesArray[index].position) - 1;
  } 
  
  return Number(bellowStory.position) + Number(aboveStory.position) / 2;
}
```

Reply:
I don't get it. In the second `if` i'm checking if the `aboveStory` is `undefined` and in the first `if` i'm checking if `bellowStory` is `undefined`.

Reply:
Sorry, I said `aboveStory` wrongly. I mean `bellowStory`

First if -> `bellowStory === undefined`
  - **`bellowStory` is `undefined`**
Sedond if -> `aboveStory !== undefined && bellowStory !== undefined`
  - `aboveStory` is not `undefined`
  - **`bellowStory` is not `undefined`**

The is no point check if `bellowStory` is not `undefined` because, if `bellowStory` is `undefined` you will reach the first if and never the second one.

Reply:
Yeah, makes sense. thanks